Tryouts/Core/Abstractions/ISubscriptionClient.cs

```diff
@@ -18,10 +18,13 @@ public interface ISubscriptionClient
     {
         Task Connect(string host, int port);
 
-        Task Publish<TMessage>(Topic topic, TMessage msg)
-             where TMessage : IMessage;
+        Task Publish(Topic topic, IMessage msg);
 
         IAsyncEnumerable<TMessage> Subscribe<TMessage>(Topic topic)
             where TMessage : IMessage, new();
+
+        IAsyncEnumerable<IMessage> Subscribe(Topic topic, Type messageType);
+
+        void Unsubscribe();
```

Comment:
What is this method doing? Is it unsubscribing from everything? Who's going to call this method?

Reply:
The client can call this method e.g. before the corresponding functionality is destroyed - e.g. when a docking pane is destroyed.

Reply:
I changed the API, current Subscribe returns disposable unsubscriber. Also topics are now string.

Reply:
I simplified the API further to 

     IObservable<CommunicationMessage> Subscribe(string topic)

Reply:
And I changed the method name to ConsumeTopic(string topic)